CHANGELOG.v11.md

```diff
@@ -0,0 +1,684 @@
+## 11.0.0-rc.0 (2022-7-7)
+
+### Notes
+Based on Realm JS v10.19.5: See changelog below for details on enhancements and fixes introduced between this and the previous pre release (which was based on Realm JS v10.17.0).
+
+### Breaking change
+* Removed all code related to the legacy Chrome Debugger. Please use [Flipper](https://fbflipper.com/) as debugger.
+* Model classes passed as schema to the `Realm` constructor must now extend `Realm.Object` and will no longer have their constructors called when pulling an object of that type from the database. Existing classes already extending `Realm.Object` now need to call the `super` constructor passing two arguments:
+  - `realm`: The Realm to create the object in.
+  - `values`: Values to pass to the `realm.create` call when creating the object in the database.
+* Renamed the `RealmInsertionModel<T>` type to `Unmanaged<T>` to simplify and highlight its usage.
+* Installing via NPM from a `git://` URL is no longer supported, since we removed `src` and `vendor` from the NPM bundle, to reduce size blow-up caused by files recently added to the sub-module. This will force end-users to checkout the Git repository from GitHub when building from source. ([#4060](https://github.com/realm/realm-js/issues/4060))
+
+### Enhancements
+* Adding support for Hermes on iOS & Android.
+* Class-based models (i.e. user defined classes extending `Realm.Object` and passed through the `schema` when opening a Realm), will now create object when their constructor is called:
+
+```ts
+class Person extends Realm.Object<Person> {
+  name!: string;
+
+  static schema = {
+    name: "Person",
+    properties: { name: "string" },
+  };
+}
+
+const realm = new Realm({ schema: [Person] });
+realm.write(() => {
+  const alice = new Person(realm, { name: "Alice" });
+  // A Person { name: "Alice" } is now persisted in the database
+  console.log("Hello " + alice.name);
+});
+```
+
+### Fixed
+* Fixed build error (call to implicitly-deleted copy constructor of 'realm::js::RealmClass<realm::js::realmjsi::Types>::Arguments') (follow up to [#4568](https://github.com/realm/realm-js/pull/4568))
+
+### Compatibility
+* React Native equal to or above `v0.66.0` and below `v0.69.0` (not included), since we're shipping binaries pre-compiled against the JSI ABI.
+* Atlas App Services.
+* Realm Studio v12.0.0.
+* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.x series.
+* File format: generates Realms with format v22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).
+
+### Internal
+* Remove the previous implementation to the JavaScriptCore engine (in `src/jsc`).
+* Upgrade Example to RN v0.68.2
+* Upgrade dependencies of the Realm Web integration tests
+* Throw instances of `Error` instead of plain objects on app errors.
+* Make integration tests on React Native Android connect to host machine by default
+
+## 10.20.0-beta.5 (2022-4-13)
```

Comment:
There are a few occurrences of `10.20` in this file. Move to `CHANGELOG.v10.md`?

Reply:
I should have mentioned that. Originally we aimed for v10.20.0 but ended with v11.0.0. I included v10.20.x in `CHANGELOG.v11.md` as I see these releases as prereleases for v11. I'll give you the long story another day 🙈.

Reply:
Haha okay 😄